src/main/daml/Daml/Finance/Data/Observable/Observation.daml

```diff
@@ -43,4 +43,4 @@ template Observation
       view = Disclosure.View with disclosureControllers = singleton provider; observers
       setObservers Disclosure.SetObservers{newObservers} = toInterfaceContractId <$>
         create this with observers = newObservers
-      archive' self = archive (coerceContractId self : ContractId Observation)
+      archive' = archive . fromInterfaceContractId @Observation
```

Comment:
😄 and 😢 .

The 😢 comes from me having to solve a bunch of merge conflicts on my draft PR :)

Reply:
Sorry!